### PR TITLE
Java Form and DynamicForm cleanup

### PR DESCRIPTION
**Pull request by mkurz**
Preparations for an upcoming pull requests that deprecates `Http.Context` related stuff inside Java Forms.

Reviewing [per commit](https://github.com/playframework/playframework/pull/8731/commits) should be the easiest.

### CODE REVIEW

**Comment by mkurz** [comment]
Rebased because of #8728

File: framework/src/play-java-forms/src/main/java/play/data/DynamicForm.java [hunk 1]
```diff
@@ -146,47 +145,45 @@ public DynamicForm bind(Map<String,String> data, String... allowedFields) {
         );
     }
 
-    /**
-     * Retrieve an error by key.
-     *
-     * @deprecated Deprecated as of 2.7.0. Method has been renamed to {@link #error(String)}.
-     */
-    @Deprecated
-    public Optional<ValidationError> getError(String key) {
```
Comment (marcospereira) [inline framework/src/play-java-forms/src/main/java/play/data/DynamicForm.java]: Not remove this?

Comment (mkurz) [reply framework/src/play-java-forms/src/main/java/play/data/DynamicForm.java]: I think it's ok to remove it. It just calls `error(key)` which we override in the subclass.

Comment (marcospereira) [reply framework/src/play-java-forms/src/main/java/play/data/DynamicForm.java]: Yeah, but we are then removing without deprecating.

Comment (mkurz) [reply framework/src/play-java-forms/src/main/java/play/data/DynamicForm.java]: But it's deprecated in the parent class already.

Comment (mkurz) [reply framework/src/play-java-forms/src/main/java/play/data/DynamicForm.java]: Here:
https://github.com/playframework/playframework/blob/be0e28c1e72be0ba02101e09852c3347835d986f/framework/src/play-java-forms/src/main/java/play/data/Form.java#L649-L658

Comment (marcospereira) [reply framework/src/play-java-forms/src/main/java/play/data/DynamicForm.java]: Got it. It is just removing the override, right?

Comment (mkurz) [reply framework/src/play-java-forms/src/main/java/play/data/DynamicForm.java]: Exactly.

**Pull request merged**